### PR TITLE
Docs: Add `WRITE LOCALLY ORDERED BY` and `WRITE DISTRIBUTED BY` in spark-ddl.md

### PR DESCRIPTION
**Pull request by xiaotianzhang01**
Enrich the description of the syntax of `write distribution and ordering` based on alter table.

### CODE REVIEW

**Comment by xiaotianzhang01** [comment]
In addition, I would like to know whether the partition field can be set by WRITE DISTRIBUTED BY, how we designed it

File: site/docs/spark-ddl.md [hunk 1]
```diff
@@ -360,3 +360,29 @@ ALTER TABLE prod.db.sample WRITE ORDERED BY category ASC NULLS LAST, id DESC NUL
...
+```sql
+ALTER TABLE prod.db.sample WRITE LOCALLY ORDERED BY category, id
+-- use optional ASC/DEC keyword to specify sort order of each field (default ASC)
+ALTER TABLE prod.db.sample WRITE LOCALLY ORDERED BY category ASC, id DESC
+-- use optional NULLS FIRST/NULLS LAST keyword to specify null order of each field (default FIRST)
+ALTER TABLE prod.db.sample WRITE LOCALLY ORDERED BY category ASC NULLS LAST, id DESC NULLS FIRST
+```
+### `ALTER TABLE ... WRITE DISTRIBUTED BY PARTITION` 
+
+Iceberg tables can be configured with a hash distribution where tuples that share the same values for clustering expressions are
```
Comment (rdblue) [inline site/docs/spark-ddl.md:374]: The requirement is to distribute by partition. Hash distribution is an implementation detail. Instead, I think this should state that `WRITE DISTRIBUTED BY PARTITION` will guarantee that a given partition is handled by one writer.

File: site/docs/spark-ddl.md [hunk 2]
```diff
@@ -360,3 +360,29 @@ ALTER TABLE prod.db.sample WRITE ORDERED BY category ASC NULLS LAST, id DESC NUL
 !!! Note
     Table write order does not guarantee data order for queries. It only affects how data is written to the table.
 
+Only local sorting can be set at the same time, use `LOCALLY ORDERED BY`
```
Comment (rdblue) [inline site/docs/spark-ddl.md:363]: This should first state that `WRITE ORDERED BY` sets a global ordering where rows are ordered across tasks, like using `ORDER BY` in an `INSERT` command. Then introduce `LOCALLY ORDERED BY` to order within each task but not across tasks.

File: site/docs/spark-ddl.md (hunk 1 above)
Comment (xiaotianzhang01) [reply site/docs/spark-ddl.md]: okey

File: site/docs/spark-ddl.md (hunk 2 above)
Comment (xiaotianzhang01) [reply site/docs/spark-ddl.md]: update

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Thanks, @xiaotianzhang01!

**Pull request merged**